src/index.js

```diff
@@ -1,25 +1,73 @@
 import { ApolloServer, gql } from "apollo-server-express";
+import "dotenv/config";
 import express from "express";
 import mongoose from "mongoose";
 import resolvers from "./resolvers";
 import typeDefs from "./types";
+import session from "express-session";
+import connectRedis from "connect-redis";
+import Redis from "ioredis";
+import cors from "cors";
 
 const startServer = async () => {
   const app = express();
 
+  const RedisStore = connectRedis(session);
+
+  const redis = new Redis();
+
+  app.set("trust proxy", 1);
+
+  // Temporary dynamic origin
+  var whitelist = ["https://studio.apollographql.com", "http://localhost:3000"];
+  var corsOptions = {
+    origin: function (origin, callback) {
+      if (whitelist.indexOf(origin) !== -1) {
+        callback(null, true);
+      } else {
+        callback(new Error("Not allowed by CORS"));
+      }
+    },
+    credentials: true,
+  };
+
+  app.use(cors(corsOptions));
+
+  app.use(
+    session({
+      name: "obfc",
+      store: new RedisStore({
+        client: redis,
+        disableTouch: true,
+      }),
+      cookie: {
+        maxAge: 1000 * 60 * 60 * 24 * 365 * 10, // 10 years
```

Comment:
What’s the use of maxAge in cookie?

Reply:
It defines how long the cookie will stay active.

![image](https://user-images.githubusercontent.com/59534570/178115231-95f159e2-04f8-4ab6-ba00-c49645291100.png)